freqtrade/freqtradebot.py

```diff
@@ -647,10 +648,38 @@ def handle_stoploss_on_exchange(self, trade: Trade) -> bool:
                 trade.sell_reason = SellType.STOPLOSS_ON_EXCHANGE.value
                 trade.update(order)
                 result = True
-            else:
-                result = False
+            elif self.config.get('trailing_stop', False):
+                # if trailing stoploss is enabled we check if stoploss value has changed
+                # in which case we cancel stoploss order and put another one with new
+                # value immediately
+                self.handle_trailing_stoploss_on_exchange(trade, order)
+
         return result
 
+    def handle_trailing_stoploss_on_exchange(self, trade: Trade, order):
+        """
+        Check to see if stoploss on exchange should be updated
+        in case of trailing stoploss on exchange
+        :param Trade: Corresponding Trade
+        :param order: Current on exchange stoploss order
+        :return: None
+        """
+
+        if trade.stop_loss > float(order['info']['stopPrice']):
+            # we check if the update is neccesary
+            update_beat = self.strategy.order_types['stoploss_on_exchange_interval']
+            if (datetime.utcnow() - trade.stoploss_last_update).total_seconds() > update_beat:
+                # cancelling the current stoploss on exchange first
+                logger.info('Trailing stoploss: cancelling current stoploss on exchange '
+                            'in order to add another one ...')
+                if self.exchange.cancel_order(order['id'], trade.pair):
+                    # creating the new one
+                    stoploss_order_id = self.exchange.stoploss_limit(
+                        pair=trade.pair, amount=trade.amount,
+                        stop_price=trade.stop_loss, rate=trade.stop_loss * 0.99
+                    )['id']
+                    trade.stoploss_order_id = str(stoploss_order_id)
```

Comment:
would you want to log that you were successful creating a new one ... ?

Reply:
It is logged in stoploss order already. I added a log there. Independent of trailing. whenever a stoploss on exchange added, there is a log:

```
2019-01-17 09:26:50,890 - freqtrade.freqtradebot - INFO - Trailing stoploss: cancelling current stoploss on exchange in order to add another one ...
2019-01-17 09:26:51,266 - freqtrade.exchange - INFO - stoploss limit order added for AE/ETH. stop price: 0.00371. limit: 0.003673
```

Reply:
missed that - sry